packages/components/src/Select/Select.test.tsx

```diff
@@ -1,67 +1,195 @@
+test('supports placeholder prop', () => {
+  render(
+    <ThemeProvider theme={theme}>
+      <Select label="MyLabel" placeholder="placeholder" data-testid="selectId">
+        <Item>1</Item>
+      </Select>
+    </ThemeProvider>
+  );
+  const button = screen.getByTestId('selectId');
+  expect(button).toHaveTextContent(/placeholder/);
 });
 
-test('renders correct HTML element', () => {
+test('option list opens when element is clicked', () => {
   render(
     <ThemeProvider theme={theme}>
-      <Select title="select">
-        <option>1</option>
+      <Select label="MyLabel" data-testid="selectId">
+        <Item>Red</Item>
       </Select>
     </ThemeProvider>
   );
-  const select = screen.getByTitle(/select/);
+  const button = screen.getByTestId('selectId');
+  expect(button).toHaveAttribute('aria-expanded', 'false');
 
-  expect(select instanceof HTMLSelectElement).toBeTruthy();
+  fireEvent.click(button);
+
+  // more than one item found because of the HiddenSelect component
+  const items = screen.getAllByText(/Red/);
+  expect(items[1]).toBeVisible();
+  expect(button).toHaveAttribute('aria-expanded', 'true');
 });
 
-test('accepts custom styles prop className', () => {
+test('supports click and select an option', () => {
   render(
     <ThemeProvider theme={theme}>
-      <Select className="custom-class-name" title="select">
-        <option>1</option>
+      <Select label="MyLabel" data-testid="selectId">
+        <Item>Red</Item>
       </Select>
     </ThemeProvider>
   );
-  const select = screen.getByTitle(/select/);
+  const button = screen.getByTestId('selectId');
+  expect(button).toHaveAttribute('aria-expanded', 'false');
 
-  expect(select.className).toMatch('custom-class-name');
+  fireEvent.click(button);
+  expect(button).toHaveAttribute('aria-expanded', 'true');
+  const items = screen.getAllByText(/Red/);
+  expect(items[1]).toBeVisible();
+  expect(items[1]).toHaveAttribute('aria-selected', 'false');
+
+  fireEvent.click(items[1]);
+  expect(button).toHaveTextContent('Red');
+
+  fireEvent.click(button);
+
+  // after selecting one item there are three elements with item text
+  const newItems = screen.getAllByText(/Red/);
+  expect(newItems[2]).toHaveAttribute('aria-selected', 'true');
+});
+
+test('popup closes after an option is selected', () => {
+  render(
+    <ThemeProvider theme={theme}>
+      <Select label="MyLabel" data-testid="selectId">
+        <Item>Red</Item>
+      </Select>
+    </ThemeProvider>
+  );
+  const button = screen.getByTestId('selectId');
+  expect(button).toHaveAttribute('aria-expanded', 'false');
+
+  fireEvent.click(button);
+  expect(button).toHaveAttribute('aria-expanded', 'true');
+  const items = screen.getAllByText(/Red/);
+  expect(items[1]).toBeVisible();
+
+  fireEvent.click(items[1]);
+  expect(button).toHaveTextContent('Red');
+  expect(items[1]).not.toBeVisible();
+});
+
+test('component has hidden Items from HiddenSelect component and displayed Items', () => {
```

Comment:
What do you test here?

Reply:
That HiddenSelect works and the displayed items visible. This test is not absolutely necessary